### PR TITLE
[On hold] Make Dockerfile Self containing

### PR DESCRIPTION
**Pull request by JesterOrNot**

### CODE REVIEW

**Comment by JesterOrNot** [comment]
@RDIL thoughts

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
+FROM golang:stretch AS build
 
 RUN curl -s "https://raw.githubusercontent.com/gosh-terminal/gosh/master/tools/install" | sh
 
+FROM ubuntu
 
-CMD [ "gosh", "-v" ]
+COPY --from=build /go/bin/gosh /home/go/bin/gosh
+COPY --from=build /go/bin/history.txt /home/go/bin/history.txt
+ENV GOPATH=/home/go \
```
Comment (RDIL) [inline Dockerfile]:
```suggestion

ENV GOPATH=/home/go \
```

File: Dockerfile [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
+    PATH=$PATH:~$GOPATH/bin
+CMD [ 'gosh', '-v' ]
```
Comment (RDIL) [inline Dockerfile]:
```suggestion

CMD [ 'gosh', '-v' ]
```

**Comment by JesterOrNot** [comment]
Try opening in gitpod I changed the .gitpod.yml to use this base image

**Comment by RDIL** [comment]
@JesterOrNot I mean it builds, but it doesn't start gosh and the gosh command isn't being found

**Comment by JesterOrNot** [comment]
@RDIL yea I think the path thing is acting weird is there anything wrong with $PATH that you can see?

**Comment by RDIL** [comment]
@JesterOrNot does the gosh binary executable being written have go runtime built into it?

**Comment by JesterOrNot** [comment]
@RDIL I'm not sure what you mean can you be more specific?

**Comment by RDIL** [comment]
Does go as a programming language require go to be installed on the production system?

**Comment by RDIL** [comment]
The system where the binary is run*

**Comment by JesterOrNot** [comment]
@RDIL No I don't think so I know that the Binary has some external dependencies I just thought those were packed into the Binary the only reason it goes to $HOME/go is because I haven't gotten it to live in /usr/bin but no technically one wouldn't need anything to run the project assuming they did it manually

**Comment by RDIL** [comment]
Well so layers 2:2 and 2:3 are copying into a directory that don't exist which is probably why this is happening.  What happens when you try to move it into `usr/bin`?

**Comment by JesterOrNot** [comment]
Right now the history is implemented by a file and parts of the program look in $GOPATH/bin for that history.txt file Ideally we want to make a standard place that everyone has and have the program look there. It doesn't matter where the binary file is as long as it is within the $PATH

File: Dockerfile [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
-WORKDIR /tmp/
+RUN curl -s "https://raw.githubusercontent.com/gosh-terminal/gosh/master/tools/install" | sh
 
+FROM ubuntu
 
-RUN curl -s "https://raw.githubusercontent.com/gosh-terminal/gosh/master/tools/install" | sh
+COPY --from=build /go/bin/gosh /home/go/bin/gosh
+COPY --from=build /go/bin/history.txt /home/go/bin/history.txt
 
+ENV GOPATH=/home/go \
```
Comment (RDIL) [inline Dockerfile]:
```suggestion
```

File: Dockerfile [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
+    PATH=$PATH:~$GOPATH/bin
```
Comment (RDIL) [inline Dockerfile]:
```suggestion
ENV PATH=$HOME:$PATH
```

File: Dockerfile [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
-    PATH=$PATH:$GOROOT/bin:$GOPATH/bin
-
-WORKDIR /tmp/
+RUN curl -s "https://raw.githubusercontent.com/gosh-terminal/gosh/master/tools/install" | sh
 
+FROM ubuntu
 
-RUN curl -s "https://raw.githubusercontent.com/gosh-terminal/gosh/master/tools/install" | sh
+COPY --from=build /go/bin/gosh /home/go/bin/gosh
+COPY --from=build /go/bin/history.txt /home/go/bin/history.txt
```
Comment (RDIL) [inline Dockerfile]:
```suggestion
COPY --from=build /go/bin/history.txt $HOME/history.txt
```

File: Dockerfile [hunk 6]
```diff
@@ -1,12 +1,13 @@
...
-ENV GOPATH=$HOME/go \
-    PATH=$PATH:$GOROOT/bin:$GOPATH/bin
-
-WORKDIR /tmp/
+RUN curl -s "https://raw.githubusercontent.com/gosh-terminal/gosh/master/tools/install" | sh
 
+FROM ubuntu
 
-RUN curl -s "https://raw.githubusercontent.com/gosh-terminal/gosh/master/tools/install" | sh
+COPY --from=build /go/bin/gosh /home/go/bin/gosh
```
Comment (RDIL) [inline Dockerfile]:
```suggestion
COPY --from=build /go/bin/gosh /usr/bin/gosh
```

**Comment by RDIL** [comment]
Honestly I say we wait until we can fix the internals before merging

**Comment by JesterOrNot** [comment]
@RDIL So should we use this PR to fix the internals?

**Comment by RDIL** [comment]
No, its way too cluttered

**Comment by JesterOrNot** [comment]
So should we start fresh on a new Branch?

**Comment by RDIL** [comment]
Yeah, I got it

**Comment by JesterOrNot** [comment]
@RDIL where is the new branch?

**Comment by RDIL** [comment]
@JesterOrNot it’s called reeces-workspace

**Pull request closed without merging**